crates/noirc_frontend/src/hir/def_collector/dc_crate.rs

```diff
@@ -53,11 +56,15 @@ pub struct DefCollector {
     pub(crate) collected_functions: Vec<UnresolvedFunctions>,
     pub(crate) collected_types: HashMap<StructId, UnresolvedStruct>,
     pub(crate) collected_globals: Vec<UnresolvedGlobal>,
-    /// collected impls maps the type name and the module id in which
-    /// the impl is defined to the functions contained in that impl
-    pub(crate) collected_impls: HashMap<(Path, LocalModuleId), Vec<UnresolvedFunctions>>,
+    pub(crate) collected_impls: ImplMap,
 }
 
+/// collected impls maps the type and the module id in which
+/// the impl is defined to the functions contained in that impl
+/// along with the generics declared on the impl itself.
+type ImplMap =
+    HashMap<(UnresolvedType, LocalModuleId), Vec<(UnresolvedGenerics, Span, UnresolvedFunctions)>>;
```

Comment:
Your comment above mentions we are mapping to the functions contained in the impl and the generics. Could you clarify what `Span` is in this comment as well? I know it is the type span but just for clarity sake.

Reply:
Looking at this again, perhaps it'd be clearer to just stick the whole `NoirImpl` struct in there as a value. The only extra bit we would be cloning is the object type but it would be much clearer where all these values come from.

Reply:
Nevermind, tried it out and realized the functions are in a different form. Went ahead and added the Span comment instead.